src/core_plugins/metrics/public/components/vis_types/timeseries/series.js

```diff
@@ -0,0 +1,175 @@
+      let optionsClassname = 'kbnTabs__tab';
+      if (selectedTab === 'metrics') metricsClassName += '-active';
+      if (selectedTab === 'options') optionsClassname += '-active';
+      let seriesBody;
+      if (selectedTab === 'metrics') {
+        const handleSort = (data) => {
+          const metrics = data.map(id => model.metrics.find(m => m.id === id));
+          this.props.onChange({ metrics });
+        };
+        seriesBody = (
+          <div>
+            <Sortable
+              style={{ cursor: 'default' }}
+              dynamic={true}
+              direction="vertical"
+              onSort={handleSort}
+              sortHandle="vis_editor__agg_sort">
+              { aggs }
+            </Sortable>
+            <div className="vis_editor__agg_row">
+              <div className="vis_editor__agg_row-item">
+                <Split
+                  onChange={this.props.onChange}
+                  fields={fields}
+                  panel={panel}
+                  model={model}/>
+              </div>
+            </div>
+          </div>
+        );
+      } else {
+        seriesBody = (
+          <SeriesConfig
+            fields={this.props.fields}
+            model={this.props.model}
+            onChange={this.props.onChange} />
+        );
+      }
+      body = (
+        <div className="vis_editor__series-row">
+          <div className="kbnTabs sm">
+            <div className={metricsClassName}
+              onClick={e => this.props.switchTab('metrics')}>Metrics</div>
+            <div className={optionsClassname}
+              onClick={e => this.props.switchTab('options')}>Options</div>
+          </div>
+          {seriesBody}
+        </div>
+      );
+    }
+
+    let colorPicker;
+    if (this.props.colorPicker) {
+      colorPicker = (
+        <ColorPicker
+          disableTrash={true}
+          onChange={this.props.onChange}
+          name="color"
+          value={model.color}/>
+      );
+    }
+
+    let dragHandle;
+    if (!this.props.disableDelete) {
+      dragHandle = (
+        <Tooltip text="Sort">
+          <div className="vis_editor__sort thor__button-outlined-default sm">
+            <i className="fa fa-sort"></i>
+        </div>
+        </Tooltip>
+      );
+    }
+
+    return (
+      <div
+        className={`${this.props.className} vis_editor__series`}
+        style={this.props.style}
+        onMouseDown={this.props.onMouseDown}
+        onTouchStart={this.props.onTouchStart}>
+        <div className="vis_editor__container">
+          <div className="vis_editor__series-details">
+            <div onClick={ this.props.toggleVisible }><i className={ caretClassName }/></div>
+            { colorPicker }
+            <div className="vis_editor__row_item" style={{ display: 'flex' }}>
+              <input
+                className="vis_editor__input-grows"
+                onChange={handleFieldChange('label')}
+                placeholder='Label'
+                ref="label"
```

Comment:
In general not a fan of using uncontrolled components. I feel it's an "escape hatch", and that controlled components should be preferred unless there are very good reasons to not use them.

Also, we should try to stay away from string refs. It's very likely they will go away:

> We advise against it because string refs have some issues, are considered legacy, and are likely to be removed in one of the future releases.

See https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs

Another thing I think you could do is have an `<input name='label' ...>` and within `handleFieldChange` use `e.target.name` and `e.target.value`. That would remove a whole lot of the unnecessary `refs` use.